File: CardIndex/CardIndex/AccountModels/RegisterModel.cs
using System.ComponentModel.DataAnnotations;

namespace CardIndex.AccountModels
{
    public class RegisterModel
    {
        [Required]
        public string Email { get; set; }
        [Required]
        public string Password { get; set; }
        [Required]
        [Compare(nameof(Password))]
        public string PasswordConfirm { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string PhoneNumber { get; set; }
        public string Login { get; set; }
    }
}

File: CardIndex/CardIndex/ClientApp/src/app/models/user-models/register-model.ts
export class RegisterModel{
    Email: string;
    Password: string;
    PasswordConfirm: string;
    FirstName: string;
    LastName: string;
    PhoneNumber: string;
    Login: string;

    constructor(email?: string,
        password?: string,
        passwordConfirm?: string,
        fisrtName?: string,
        lastName?: string,
        phoneNumber?: string,
        login?: string,)
    {
        this.Email = email;
        this.Password = password;
        this.PasswordConfirm = passwordConfirm;
        this.FirstName = fisrtName;
        this.LastName = lastName;
        this.PhoneNumber = phoneNumber;
        this.Login = login;
    }
}